videostore/utils.go

```diff
@@ -287,38 +377,33 @@ func generateOutputFilePath(camName, fromStr, metadata, path string) string {
 // Extracts the start timestamp of the oldest file and the start of the most recent file.
 // Since the most recent segment file is still being written to by the segmenter
 // we do not want to include it in the time range.
-func validateTimeRange(files []string, start, end time.Time) error {
+// func validateTimeRange(files []string, start, end time.Time) error {
+func validateTimeRange(files []fileWithDate, start, end time.Time) error {
 	if len(files) == 0 {
 		return errors.New("no storage files found")
 	}
-	oldestFileStart, err := extractDateTimeFromFilename(files[0])
-	if err != nil {
-		return err
-	}
-	newestFileStart, err := extractDateTimeFromFilename(files[len(files)-1])
-	if err != nil {
-		return err
-	}
+	oldestFileStart := files[0].date
+	newestFileStart := files[len(files)-1].date
 	if start.Before(oldestFileStart) || end.After(newestFileStart) {
 		return errors.New("time range is outside of storage range")
 	}
 	return nil
 }
 
-// getVideoDuration returns the duration of the video file in seconds.
-func getVideoDuration(filePath string) (time.Duration, error) {
+// getVideoInfo calls the C function get_video_info to retrieve
+// duration, width, height, and codec of a video file.
+func getVideoInfo(filePath string) (videoInfo, error) {
```

Comment:
what does this functiuon return if called on a file that is currently being written to by the rawsegmenter?